internal/services/subscription/data_source_location.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package subscription
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	resourcesSubscription "github.com/hashicorp/go-azure-sdk/resource-manager/resources/2022-12-01/subscriptions"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ sdk.DataSource = LocationDataSource{}
+
+type LocationDataSource struct{}
+
+type LocationDataSourceModel struct {
+	Location     string                `tfschema:"location"`
+	RegionType   string                `tfschema:"region_type"`
+	DisplayName  string                `tfschema:"display_name"`
+	ZoneMappings []LocationZoneMapping `tfschema:"zone_mappings"`
+}
+
+type LocationZoneMapping struct {
+	LogicalZone  string `tfschema:"logical_zone"`
+	PhysicalZone string `tfschema:"physical_zone"`
+}
+
+func (r LocationDataSource) ResourceType() string {
+	return "azurerm_location"
+}
+
+func (r LocationDataSource) ModelObject() interface{} {
+	return &LocationDataSourceModel{}
+}
+
+func (r LocationDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.LocationWithoutForceNew(),
+	}
+}
+
+func (r LocationDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"region_type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
```

Comment:
within the azurerm provider we use `location` to mean a physical location (which supports extended locations and custom locations) - given this, we should remove this field and filter only to the physical location in the Read() function